core/Makefile

```diff
@@ -46,7 +46,7 @@ endif
 PROTOCXXFLAGS = -std=gnu++11 -O0
 
 CXXFLAGS += -std=gnu++11 -g3 -ggdb3 -Ofast -march=native \
-	    -Werror -isystem $(DPDK_INC_DIR) -D_GNU_SOURCE \
+	    -Werror -isystem $(DPDK_INC_DIR) -D_GNU_SOURCE -DNDEBUG \
```

Comment:
Perhaps we can do this as a separate section, such as:

ifndef DEBUG
CXXFLAGS += -DNDEBUG
endif